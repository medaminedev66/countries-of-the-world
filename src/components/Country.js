import React from 'react';
import { PropTypes } from 'prop-types';
import fnum from '../logic/numbers';

function Country(props) {
  const { obj } = props;

  return (
    <div className="country">
      <img className="flag-country" src={obj.flags.png} alt="flag" />
      <p className="h2 p">{obj.name}</p>
      <p className="p">
        pp:&nbsp;
        {fnum(obj.population)}
      </p>
      <p className="p">
        Region:&nbsp;
        {fnum(obj.region)}
      </p>
    </div>
  );
}

Country.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  obj: PropTypes.object.isRequired,
};

export default Country;
