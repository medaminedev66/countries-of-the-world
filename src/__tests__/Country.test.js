import React from 'react';
import Country from '../components/Country';
import { ReduxRenderer, renderWithRedux } from './Mocks/reduxStore';

describe('test the Country component', () => {
  const country = {
    name: 'Morocco',
    flags: 'flag',
  };
  it('matches the snapshot', () => {
    const tree = ReduxRenderer(<Country obj={country} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const { getByText } = renderWithRedux(<Country obj={country} />);
    expect(getByText('Morocco')).not.toBeNull();
  });
});
