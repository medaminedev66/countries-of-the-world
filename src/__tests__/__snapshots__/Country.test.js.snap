// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the Country component matches the snapshot 1`] = `
<div
  className="country"
>
  <img
    alt="flag"
    className="flag-country"
  />
  <p
    className="h2 p"
  >
    Morocco
  </p>
  <p
    className="p"
  >
    pp: 
  </p>
  <p
    className="p"
  >
    Region: 
  </p>
</div>
`;
